src/test/java/org/mockito/internal/stubbing/defaultanswers/ReturnsSmartNullsTest.java

```diff
@@ -85,4 +90,36 @@ public void should_print_the_parameters_on_SmartNullPointerException_message() t
                 .hasMessageContaining("lumpa");
         }
     }
+
+    // https://github.com/mockito/mockito/issues/357
+    @Test
+    public void generic_return_type_erasure() throws Throwable {
+        class S<T> {
+            public T f() {
+                return null;
+            }
+        }
+
+        S<Boolean> s = Mockito.mock(S.class, org.mockito.Answers.RETURNS_SMART_NULLS);
```

Comment:
I think the TypeToken/TypeLiteral approach might be better, especially for more complicated types:

```
mockGeneric(new TypeToken<S<Boolean>>() {}, RETURNS_SMART_NULLS)
```